### PR TITLE
Hotfix/fixed problems in activefacetmap

### PR DESCRIPTION
**Pull request by alexhuang091**
current activeFacetMap is a Map<String, Facet> which means different fqs can't have the same key. for example, it's possible to construct such a request from ala-hub

`&fq=-(month:"11" OR month:"12")&fq=-(month:"09" OR month:"10")&fq=-(month:"07" OR month:"08")&fq=month:"02"`, returned data is like below

```
    "activeFacetMap": {
        "month": {
            "name": "month",
            "displayName": "Month:\"February\"",
            "value": "\"02\""
        },
        "-(month": {
            "name": "-(month",
            "displayName": "-(month:\"07\" OR Month:\"August\")",
            "value": "\"07\" OR month:\"08\")"
        }
    }
```

few problems in this result: 

- ` name: "-(month"` this is wrong, it should be the key '-month'
- `"displayName": "-(month:\"07\" OR Month:\"August\")"` display name is not completely translated, 07 should be July -- **this will be fixed by changes in #437 once it's deployed into production**
- `"value": "\"07\" OR month:\"08\")"` should be `"value": "\"07\" OR\"08\")"`
- **most importantly**: there are 4 fqs, but in this map, there are only 2. This is because the first 3 have same key '-(month' so only the last one got stored.

With changes in this PR, all fqs with the same key will be stored in a list, so same request yields below response. One thing to note is in the `value` field, original fq is used. It makes ala-hub front end work easier.

```
    "activeFacetObj": {
        "month": [
            {
                "name": "month",
                "displayName": "Month:\"February\"",
                "value": "month:\"02\""
            }
        ],
        "-month": [
            {
                "name": "-month",
                "displayName": "-(Month:\"November\" OR Month:\"December\")",
                "value": "-(month:\"11\" OR month:\"12\")"
            },
            {
                "name": "-month",
                "displayName": "-(Month:\"September\" OR Month:\"October\")",
                "value": "-(month:\"09\" OR month:\"10\")"
            },
            {
                "name": "-month",
                "displayName": "-(Month:\"July\" OR Month:\"August\")",
                "value": "-(month:\"07\" OR month:\"08\")"
            }
        ]
    }
```

### CODE REVIEW

**Comment by alexhuang091** [comment]
Actually, there are quite a few existing issues related to this, example #209

File: src/test/java/au/org/ala/biocache/dao/FilterQueryParserTest.java [hunk 1]
```diff
@@ -153,22 +156,46 @@ public void testAddFacetMap6() {
...
-            Map<String, Facet> facetMap = queryFormatUtils.formatSearchQuery(query, true);
-            assertTrue(facetMap != null && facetMap.size() == 1);
-            assertTrue(facetMap.get(facetMap.keySet().iterator().next()).getDisplayName().equals(fq[1]));
+        query.setFq(fq);
+        Map<String, List<Facet>> actualResult = queryFormatUtils.formatSearchQuery(query)[1];
+        assertTrue(actualResult != null && actualResult.size() == expectedResult.size() && actualResult.keySet().equals(expectedResult.keySet()));
+
+        for (Map.Entry<String, List<Facet>> entry : actualResult.entrySet()) {
+            // Here we only test if filter keys are correctly parsed from fqs
+            assertTrue(entry.getValue().size() == expectedResult.get(entry.getKey()).size());
```
Comment (sbearcsiro) [inline src/test/java/au/org/ala/biocache/dao/FilterQueryParserTest.java:198]: Can we add a check that the expected values equal the calculated values as well?  Hamcrest provides a matcher for lists in any order

File: src/main/java/au/org/ala/biocache/util/QueryFormatUtils.java [hunk 2]
```diff
@@ -173,7 +189,52 @@ public void setMaxBooleanClauses(int maxBooleanClauses) {
...
+     * fq = key:val or fq = -key:val
+     * fq = (key:val) or fq = (-key:val)
+     * fq = -(key:val) or fq = -(-key:val)
+     * fq = key:val1 OR key:val2 or fq = -key:val1 OR -key:val2
+     * fq = (key:val1 OR key:val2) or fq = (-key:val1 OR -key:val2)
+     * fq = -(key:val1 OR key:val2) or fq = -(-key:val1 OR -key:val2)
+     *
+     * inclusive and exclusive keys can't co-exist in one fq
+     */
+    private String[] parseFQ(String fq) {
```
Comment (sbearcsiro) [inline src/main/java/au/org/ala/biocache/util/QueryFormatUtils.java:208]: Only the first array element gets used from the result of this method, might be worth returning a single `String`

Comment (alexhuang091) [reply src/main/java/au/org/ala/biocache/util/QueryFormatUtils.java:208]: fixed

File: src/test/java/au/org/ala/biocache/dao/FilterQueryParserTest.java (hunk 1 above)
Comment (alexhuang091) [reply src/test/java/au/org/ala/biocache/dao/FilterQueryParserTest.java:198]: done

File: src/test/java/au/org/ala/biocache/dao/FilterQueryParserTest.java [hunk 3]
```diff
@@ -153,22 +156,46 @@ public void testAddFacetMap6() {
     }
 
     @Test
-    public void testFqFormat() {
```
[the remark this answers is not in the document]
Comment (alexhuang091) [reply src/test/java/au/org/ala/biocache/dao/FilterQueryParserTest.java]: restored

**Pull request merged**